jujugui/static/gui/src/app/app.js

```diff
@@ -854,6 +858,46 @@ YUI.add('juju-gui', function(Y) {
     },
 
     /**
+      Callback handler for the api loginWithMacaroon method which hanadles
+      the "redirection required" error message.
+
+      @method _apiLoginHandler
+      @param {Object} api The API that the user is attempting to log into.
+      @param {String} err The login error message, if any.
+      @param {Object} data The login request response data, if any.
+    */
+    _apiLoginHandler: function(api, err, data) {
+      const errorNotify = function(err) {
+        this.db.notifications.add({
+          title: 'Unable to log into model',
+          message: `unable to log into model: ${err}`,
+          level: 'error'
+        });
+      };
+
+      if (err === 'authentication failed: redirection required') {
+        // If the error is that redirection is required then we have to
+        // make a request to get the appropriate model connection information.
+        api.redirectInfo((err, servers) => {
+          if (err) {
+            errorNotify(err);
+            return;
+          }
+          // Loop through the available servers and find the public IP.
+          const server = servers[0].filter(
+            server => server.scope === 'public');
```

Comment:
Good idea, unfortunately because JIMM is down atm I am not able to actually test this so I'll add this as a follow-up